Mb/development (#18)
hostmobility/documentation: docs/system/legacy-build-system.md

sudo docker run -it -u <b>1000</b> --privileged -v ~/.ssh:/home/<b>username</b>/.ssh -v ~/project:/home/<b>username</b>/<b>project</b> hostmobility/mx4
```

### Example to setup environment for MX-4 CT, branch BSP-v1.5.x

#### Git clone from repository

```bash
# If SSH key is associated correctly with your account on github, run the following commands:
git clone git@github.com:hostmobility/mx4 -b mx4-bsp-v1.5.x
git clone git@github.com:hostmobility/mx4-pic -b mx4-bsp-v1.5.x mx4/pic
# If not correctly associated, run follow commands:
git clone https://github.com/hostmobility/mx4/ -b mx4-bsp-v1.5.x
git clone https://github.com/hostmobility/mx4-pic -b mx4-bsp-v1.5.x mx4/pic
```
#### Build CT-BSP-v1.5.x

```bash
cd mx4/t20
# Remember to edit username to your own username and project if you
# named your project folder to something else.
# edit $(nproc) to number of cores you want to use while building.
./make_system.sh -t ct -r /home/<b>username</b>/<b>project</b>/rootfs/CT -d /home/<b>username</b>/<b>project</b>/yocto-1.5.x -g -k -u -j <b>$(nproc)</b> -m -T 512
```

### Building other platforms

> The needed repositories and branch names can be seen here: https://github.com/hostmobility/mx4#source-structure

Reviewer: ```suggestion
Please consult [mx4#source-structure](https://github.com/hostmobility/mx4#source-structure) for the required repositories and branch names. 
```

### GTT v1.5.x

```bash
git clone https://github.com/hostmobility/mx4/ -b mx4-bsp-v1.5.x
git clone https://github.com/hostmobility/mx4-pic -b mx4-bsp-v1.5.x mx4/pic
./make_system.sh -t gtt -r <b>UNIQUE_ROOTFS_PATH</b> -d <b>YOCTO_1.5.x_TEMP_PATH</b> -g -k -u -j <b>$(nproc)</b> -m -T 512
```

### GTT-T30 v1.4.x

```bash
git clone https://github.com/hostmobility/mx4/ -b mx4-bsp-v1.4.x-ultra
git clone https://github.com/hostmobility/mx4-pic -b mx4-bsp-v1.4.x mx4/pic
git clone https://github.com/hostmobility/linux-toradex.git -b mx4-bsp-v1.4.x-tegra mx4/t20/linux_vf
git clone https://github.com/hostmobility/u-boot-toradex.git -b mx4-bsp-v1.4.x mx4/t20/u-boot_vf
./make_system.sh -t t30 -r UNIQUE_ROOTFS_PATH -d YOCTO_1.4.x_TEMP_PATH -g -k -u -j $(nproc)
```

### Example to setup environment for MX-4 V61
```bash
# All products use mx4 and mx4-pic repository
git clone git@github.com:hostmobility/mx4 -b mx4-2.0
git clone git@github.com:hostmobility/mx4-pic -b mx4-2.0 mx4/pic

> # Below are only required for products who have Linux and U-boot

Reviewer: ```suggestion
# Below are only required for products that have Linux and U-boot
```

# outside of the mx4 repository
git clone git@github.com:hostmobility/linux-toradex.git -b hm_vf_4.4 mx4/t20/linux_vf
git clone git@github.com:hostmobility/u-boot-toradex.git -b 2015.04-hm mx4/t20/u-boot_vf
```

> ### Example to build Board Support Package for MX-4 V61

Reviewer: ```suggestion
### Example of how to build Board Support Package (BSP) for MX-4 V61
```

```bash
cd mx4/t20
# edit $(nproc) to number of cores you want to use while building.
[work in progress]
./make_system.sh -t v61 -r /home/username/project/rootfs/v61 -d /home/username/project/ -u -k -j $(nproc) -g
```

> List of supported distros can be found at http://www.yoctoproject.org/docs/1.4.1/ref-manual/ref-manual.html#detailed-supported-distros

Reviewer: ```suggestion
See the [List of supported distros](http://www.yoctoproject.org/docs/1.4.1/ref-manual/ref-manual.html#detailed-supported-distros).

> Following packages need to be installed. Below example is Ubuntu 14.10

Reviewer: ```suggestion
The following packages need to be installed (the example below is for Ubuntu 14.10).
```
